cumulusci/tasks/salesforce/Deploy.py

```diff
@@ -42,29 +42,56 @@ class Deploy(BaseSalesforceMetadataApiTask):
         },
     }
 
+    # ignore all files in ./lwc and all sub-directories of a lwc component (i.e. ./lwc/[lwc Component Name]/.*)
+    _directory_blacklist = re.compile("^\\.\\/lwc(\\/.*?\\/.*)?$")
+
+    def _include_directory(self, root):
+        return not self._directory_blacklist.match(root)
+
+    # ignore all lwc component sub-directories (usually __tests__ if anything)
+    _lwc_component_pattern = re.compile("^\\.\\/lwc\\/.*$")
+
+    def _is_lwc_component(self, root):
+        return self._lwc_component_pattern.match(root)
+
+    # whitelist all files ending in one of the following: .js, .js-meta.xml, .html, .css, .svg
+    _lwc_component_file_extension_whitelist = re.compile(
+        "^.*\\.(js(-meta\\.xml)?|html|css|svg)$"
+    )
+
+    def _include_lwc_component_file(self, f):
+        return self._lwc_component_file_extension_whitelist.match(f)
+
+    def _get_api(self, path=None):
+        if not path:
+            path = self.task_config.options__path
+
+        package_zip = self._get_package_zip(path)
+        self.logger.info("Payload size: {} bytes".format(len(package_zip)))
+        return self.api_class(self, package_zip, purge_on_delete=False)
+
     def _get_package_zip(self, path):
         # Build the zip file
         zip_bytes = io.BytesIO()
         zipf = zipfile.ZipFile(zip_bytes, "w", zipfile.ZIP_DEFLATED)
 
         with cd(path):
             for root, dirs, files in os.walk("."):
-                for f in files:
-                    self._write_zip_file(zipf, root, f)
+                if self._include_directory(root):
+                    if self._is_lwc_component(root):
+                        for f in files:
+                            if self._include_lwc_component_file(f):
+                                self._write_zip_file(zipf, root, f)
+                    else:
+                        # Include all files
+                        for f in files:
+                            self._write_zip_file(zipf, root, f)
```

Comment:
I'd like to have all the filtering logic in a single method because it will make both the directory-walking logic and the filtering logic clearer (and easier to test) if they are separate. I'm anticipating that we may need to add different filters for other component types in the future, and that will be cleanest if things are sufficiently decoupled that we can delegate the logic for each component to a different class.

I also think it's going to be hard to make this work using regular expressions on Windows which uses a backslash as a path separator. How about something like this:

    for root, dirs, files in os.walk("."):
        root_parts = root.split(os.sep)[1:]
        for f in files:
            parts = root_parts + f.split(os.sep)
            if not self._include_file(parts):
                continue
            self._write_zip_file(zipf, root, f)

And then `_include_file` can work with a list of path segments rather than a string.

Reply:
-  I will decouple the logic into separate methods.
-  I'm surprised using regex is really a compatibility issue (and it makes sense that it is).  I like your solution and will implement

Reply:
It's not the regex per se, it's the fact that the patterns aren't accounting for varying path separators. I haven't thought through the details of what making assertions against a list of path elements will look like, so tell me if it turns out to not make sense.

Reply:
I created methods `_include_directory` and `_include_file`.  The methods don't interact with the classes state at all.  Can they be static methods, or is it preferred I leave them as instance methods?

Reply:
@spelak-salesforce let's leave them as instance methods. That way if we need to refer to the task options at some point to determine what to include, that will be possible.